tests/nosetests/pyanaconda_tests/product_test.py

```diff
@@ -0,0 +1,257 @@
+
+        config.validate()
+
+    def fedora_products_test(self):
+        self._check_default_product(
+            "Fedora", "",
+            ["fedora.conf"]
+        )
+        self._check_default_product(
+            "Fedora", "Server",
+            ["fedora.conf", "fedora-server.conf"]
+        )
+        self._check_default_product(
+            "Fedora", "Workstation",
+            ["fedora.conf", "fedora-workstation.conf"]
+        )
+        self._check_default_product(
+            "Fedora", "Atomic Host",
+            ["fedora.conf", "fedora-atomic-host.conf"]
+
+        )
+        self._check_default_product(
+            "Fedora", "Silverblue",
+            ["fedora.conf", "fedora-workstation.conf", "fedora-silverblue.conf"]
+        )
+
+    def rhel_products_test(self):
+        self._check_default_product(
+            "Red Hat Enterprise Linux", "",
+            ["rhel.conf"]
+        )
+        self._check_default_product(
+            "CentOS Linux", "",
+            ["rhel.conf", "centos.conf"]
+        )
+        self._check_default_product(
+            "Scientific Linux", "",
+            ["rhel.conf", "scientific-linux.conf"]
+        )
+
+    def valid_product_test(self):
+        content = dedent("""
+        [Product]
+        product_name = My Product
+        """)
+
+        base_path = self._load_product(content)
+        self._check_product("My Product", "", [base_path])
+
+        content = dedent("""
+        [Product]
+        product_name = My Next Product
+
+        [Base Product]
+        product_name = My Product
+        """)
+
+        path = self._load_product(content)
+        self._check_product("My Next Product", "", [base_path, path])
+
+    def valid_variant_test(self):
+        content = dedent("""
+        [Product]
+        product_name = My Product
+        variant_name = My Variant
+        """)
+
+        base_path = self._load_product(content)
+        self._check_product("My Product", "My Variant", [base_path])
+
+        content = dedent("""
+        [Product]
+        product_name = My Product
+        variant_name = My Next Variant
+
+        [Base Product]
+        product_name = My Product
+        variant_name = My Variant
+        """)
+
+        path = self._load_product(content)
+        self._check_product("My Product", "My Next Variant", [base_path, path])
+
+    def invalid_product_test(self):
+        with self.assertRaises(ConfigurationError):
+            self._load_product("")
+
+        with self.assertRaises(ConfigurationError):
+            self._load_product("[Product]")
+
+        with self.assertRaises(ConfigurationError):
+            self._load_product("[Base Product]")
+
+        content = dedent("""
+        [Product]
+        variant_name = Server
+
+        """)
+
+        with self.assertRaises(ConfigurationError):
+            self._load_product(content)
+
+        content = dedent("""
+        [Product]
+        variant_name = Server
+
+        """)
+
```

Comment:
This content seems to be the same as the content above ?

Reply:
Yes, it is. Fixed.